app/controllers/prison/messages_controller.rb

```diff
@@ -0,0 +1,30 @@
+class Prison::MessagesController < ApplicationController
+  before_action :authorize_prison_request
+  before_action :authenticate_user
+
+  def create
+    @visit = load_visit
+    @message = Message.create_and_send_email(message_params)
+
+    if @message.persisted?
+      flash[:notice] = t('message_created', scope: [:prison, :flash])
+      redirect_to :back
+    else
+      flash[:notice] = t('message_create_error', scope: [:prison, :flash])
+      render 'prison/visits/show'
+    end
+  end
+
+private
+
+  def load_visit
+    Visit.joins(prison: :estate).
+      where(estates: { id: current_estate }).
+      find(params[:visit_id])
+  end
+
+  def message_params
+    params.require(:message).permit(:body).
+      merge(user: current_user, visit: @visit)
+  end
+end
```

Comment:
There is a tricky bit here, ```load_visit``` needs to be called before ```message_params```, one could easily be trapped here.
I believe the following would be clearer:

```ruby
class Prison::MessagesController < ApplicationController
  before_action :authorize_prison_request
  before_action :authenticate_user

  def create
    load_visit
    @message = Message.create_and_send_email(message_params)

    if @message.persisted?
      flash[:notice] = t('message_created', scope: [:prison, :flash])
      redirect_to :back
    else
      flash[:notice] = t('message_create_error', scope: [:prison, :flash])
      render 'prison/visits/show'
    end
  end

private

  def visit
    @visit ||= Visit.joins(prison: :estate).
      where(estates: { id: current_estate }).
      find(params[:visit_id])
  end
  alias :load_visit :visit

  def message_params
    params.require(:message).permit(:body).
      merge(user: current_user, visit: visit)
  end
end
```
I usually wouldn't bother with aliasing ```load_visit``` to ```visit``` but it's a bit more explicit.